### PR TITLE
Uplift mutability to `rustc_type_ir`

### PR DESCRIPTION
**Pull request by compiler-errors**
Part of rust-lang/types-team#124

I'm less satisfied with this one compared to #116918 -- namely, there are a few places (other than astconv itself) that we have to shunt between `hir::Mutability` and `ty::Mutability`. Maybe we should have a trait in `rustc_middle` to convert between them?

r? types

### CODE REVIEW

**Comment by compiler-errors** [comment]
wait I definitely didn't check if this passed clippy or rustdoc

@rustbot author

File: compiler/rustc_const_eval/src/interpret/validity.rs [hunk 1]
```diff
@@ -533,7 +533,7 @@ impl<'rt, 'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> ValidityVisitor<'rt, 'mir, '
             }
             ty::Ref(_, ty, mutbl) => {
                 if matches!(self.ctfe_mode, Some(CtfeValidationMode::Const { .. }))
-                    && *mutbl == Mutability::Mut
+                    && *mutbl == ty::Mutability::Mut
```
Comment (RalfJung) [inline compiler/rustc_const_eval/src/interpret/validity.rs:536]: `ty::Mutability` has been imported, why is this still qualified?

Comment (compiler-errors) [reply compiler/rustc_const_eval/src/interpret/validity.rs:536]: probably did that before i realized it was easier just to change the imports

**Pull request closed without merging**